### PR TITLE
Initial packaging of gotestsum

### PR DESCRIPTION
**Pull request by mjmac**
Provides a `go test` wrapper that emits test reports in the
JUnit format.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,9 @@
+NAME	:= gotestsum
+SRC_EXT	:= gz
+VERSION	:= 0.4.0
```
Comment (brianjmurrell) [inline Makefile]: Shouldn't need this.  It will be extracted from the specfile automatically.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+
+include packaging/Makefile_packaging.mk
+
+$(VERSION).tar.gz:
+	rm -f ./$@
+	curl -f -L -O 'https://github.com/gotestyourself/$(NAME)/archive/v$(VERSION).tar.gz'
```
Comment (brianjmurrell) [inline Makefile]: There should be rules to download source artefacts in `packaging/Makefile_packaging.mk` already:

https://github.com/daos-stack/gotestsum/blob/9643e6fb35ea164332364c901120e26ee97ad62c/packaging/Makefile_packaging.mk#L127-L141

and in particular it looks like this one:

https://github.com/daos-stack/gotestsum/blob/9643e6fb35ea164332364c901120e26ee97ad62c/packaging/Makefile_packaging.mk#L135-L137

should be doing it.

The recipe here is a bit interesting though.  Your _target_ is `$(VERSION).tar.gz` yet what's being downloaded is `v$(VERSION).tar.gz`.  Typo on the recipe target perhaps?

File: Makefile (hunk 1 above)
Comment (mjmac) [reply Makefile]: Nifty, hadn't realized that the included makefile handled all this stuff.

**Comment by brianjmurrell** [comment]
You also need to copy: `Dockerfile.centos.7`, `Dockerfile.coverity`, `Dockerfile.ubuntu.18.04`, `Dockerfile.ubuntu.rolling` from the packaging repo to your `packaging/` subdir.  Also looks like you have an extra newline at the end of `packaging/Makefile_packaging.mk`.  Probably easiest to just re-copy that from the packaging repo also.

**Comment by mjmac** [comment]
Closing this in favor of the approach in #2 (packaging submodule).

**Pull request closed without merging**